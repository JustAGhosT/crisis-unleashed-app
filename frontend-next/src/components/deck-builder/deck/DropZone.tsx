import React from "react";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import { cn } from "@/lib/utils";

interface DropZoneProps {
  title: string;
  totalCount: number;
  isDragOver: boolean;
  isInvalidDrop: boolean;
  className?: string;
  onDrop: (e: React.DragEvent) => void;
  onDragOver: (e: React.DragEvent) => void;
  onDragEnter: (e: React.DragEvent) => void;
  onDragLeave: (e: React.DragEvent) => void;
  headerRight?: React.ReactNode;
  children?: React.ReactNode;
}

export const DropZone: React.FC<DropZoneProps> = ({
  title,
  totalCount,
  isDragOver,
  isInvalidDrop,
  className,
  onDrop,
  onDragOver,
  onDragEnter,
  onDragLeave,
  headerRight,
  children,
}) => {
  return (
    <Card
      className={cn(
        "bg-slate-800/50 border-slate-600 transition-colors",
        isDragOver && "border-amber-400 ring-2 ring-amber-400/50",
        isInvalidDrop && "border-red-500 ring-2 ring-red-500/50",
        className,
      )}
      onDrop={onDrop}
      onDragOver={onDragOver}
      onDragEnter={onDragEnter}
      onDragLeave={onDragLeave}
    >
      <CardHeader>
        <div className="flex justify-between items-center">
          <CardTitle className="flex items-center gap-2">
            {title}
            <Badge variant="outline" className="ml-2">
              {totalCount} cards
            </Badge>
          </CardTitle>
          {headerRight}
        </div>
      </CardHeader>
      <CardContent>{children}</CardContent>
    </Card>
  );
};

DropZone.displayName = "DropZone";
