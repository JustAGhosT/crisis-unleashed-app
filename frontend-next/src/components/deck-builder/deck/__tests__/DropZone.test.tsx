import React from "react";
import { render } from "@testing-library/react";
import { DropZone } from "../DropZone";

describe("DropZone", () => {
  it("renders header and children", () => {
    const { container, getByText } = render(
      <DropZone
        title="Current Deck"
        totalCount={5}
        isDragOver={false}
        isInvalidDrop={false}
        onDrop={jest.fn()}
        onDragOver={jest.fn()}
        onDragEnter={jest.fn()}
        onDragLeave={jest.fn()}
      >
        <div>content</div>
      </DropZone>,
    );
    expect(getByText("Current Deck")).toBeInTheDocument();
    expect(getByText("5 cards")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
