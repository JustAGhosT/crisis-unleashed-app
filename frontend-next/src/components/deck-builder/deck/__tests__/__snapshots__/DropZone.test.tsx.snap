// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropZone renders header and children 1`] = `
<div
  class="rounded-lg border text-card-foreground shadow-sm bg-slate-800/50 border-slate-600 transition-colors"
>
  <div
    class="flex flex-col space-y-1.5 p-6"
  >
    <div
      class="flex justify-between items-center"
    >
      <h3
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        Current Deck
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground ml-2"
        >
          5
           cards
        </div>
      </h3>
    </div>
  </div>
  <div
    class="p-6 pt-0"
  >
    <div>
      content
    </div>
  </div>
</div>
`;
